components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManager.java

```diff
@@ -522,4 +540,390 @@ public RolesGetResponse listRolesWithPost(SearchRequest searchRequest)
         return listRolesWithGET(searchRequest.getFilter(), searchRequest.getStartIndex(), searchRequest.getCount(),
                 searchRequest.getSortBy(), searchRequest.getSortOder());
     }
+
+    @Override
+    public Role patchRole(String roleId, Map<String, List<PatchOperation>> patchOperations)
+            throws BadRequestException, CharonException, ConflictException, NotFoundException {
+
+        String currentRoleName = getCurrentRoleName(roleId, tenantDomain);
+
+        if (log.isDebugEnabled()) {
+            log.debug("Updating Role: " + currentRoleName);
+        }
+
+        try {
+            List<PatchOperation> displayNameOperations = new ArrayList<>();
+            List<PatchOperation> memberOperations = new ArrayList<>();
+            List<PatchOperation> groupOperations = new ArrayList<>();
+            List<PatchOperation> permissionOperations = new ArrayList<>();
+
+            for (List<PatchOperation> patchOperationList : patchOperations.values()) {
```

Comment:
Add a null check for patchOperations before iterating the list. Since this is a public method, there could be a chance of passing null

Reply:
Fixed